src/defrag.c

```diff
@@ -770,17 +770,17 @@ void defragScanCallback(void *privdata, const dictEntry *de) {
  * or not, a false detection can cause the defragmenter to waste a lot of CPU
  * without the possibility of getting any results. */
 float getAllocatorFragmentation(size_t *out_frag_bytes) {
-    size_t resident, active, allocated;
-    zmalloc_get_allocator_info(&allocated, &active, &resident);
-    float frag_pct = ((float)active / allocated)*100 - 100;
-    size_t frag_bytes = active - allocated;
+    size_t resident, active, allocated, frag_smallbins_bytes;
+    zmalloc_get_allocator_info(&allocated, &active, &resident, NULL, NULL, &frag_smallbins_bytes);
+
+    float frag_pct = (float)frag_smallbins_bytes / active * 100;
```

Comment:
let's add a comment stating that it's a ratio of wasted memory specifically in small bins, as a portion of all the allocated memory (including large-bins), and explain that otherwise, when redis mostly uses large bins, even though the fragmentation doesn't consume a lot of memory, it'll show high percentage. 
also, as we discussed in the other issue, i think the divisor should be `allocated`, not `active`

Reply:
done with [`95ccec6` (#12996)](https://github.com/redis/redis/pull/12996/commits/95ccec656640a3698e1eabb72fb31611117ff4be)